modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/StoriesViewModel.kt

```diff
@@ -0,0 +1,124 @@
+package au.com.shiftyjelly.pocketcasts.endofyear
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.endofyear.stories.Story
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.util.Timer
+import javax.inject.Inject
+import kotlin.concurrent.fixedRateTimer
+import kotlin.math.roundToInt
+
+@HiltViewModel
+class StoriesViewModel @Inject constructor(
+    private val storiesDataSource: StoriesDataSource,
+) : ViewModel() {
+    private val mutableState = MutableStateFlow<State>(State.Loading)
+    val state: StateFlow<State> = mutableState
+
+    private val mutableProgress = MutableStateFlow(0f)
+    val progress: StateFlow<Float> = mutableProgress
+
+    private val numOfStories: Int
+        get() = storiesDataSource.numOfStories
+
+    private var currentIndex: Int = 0
+    private val nextIndex
+        get() = (currentIndex.plus(1)).coerceAtMost(numOfStories.minus(1))
```

Comment:
I'd like to keep the current behavior for now. I thought of starting from the beginning for this scenario but if there are several stories, it might not be a pleasant experience going through all stories. And as you mentioned, skipping to the end and having nothing else to play could also be confusing.